sunpy/io/fits.py

```diff
@@ -183,7 +185,20 @@ def write(fname, data, header, hdu_type=None, **kwargs):
     if not hdu_type:
         hdu_type = fits.PrimaryHDU
 
-    hdu = hdu_type(data=data, header=fits_header)
+    if isinstance(hdu_type, type):
+        hdu = hdu_type(data=data, header=fits_header)
+    else:  # HDU already initialised
```

Comment:
I think this is better done the other way around.

```python
if isinstance(hdu_type, (fits.PrimaryHDU, fits.ExtensionHDU)):
```
should catch all the cases of an already instantiated HDU, and that leaves the alternative a little more flexible.